### ISSUE TITLE
Some indels are not called. Any recommendation?

### ISSUE THREAD

**Issue by erfan1985** [cauyrd/ScanIndel#12]
I have tried ScanIndel on a couple of exome data files with known complex or long del/dups. I have not been able to call a few of them. This happened in situations where I expected to get a call. For instance it detected a very long deletion >10kb, but it missed a 40bp del and a 200 bp insertions as well as a 1kb deletion in the middle of a pileup. Are there any parameters that I should adjust? Any reason why I am not getting these called?
Thanks

**Comment by cauyrd** [comment]
Interesting. Can you send me the sliced bam file containing these missed
indels. I will take a test.

Rendong

**Comment by erfan1985** [comment]
I m attaching three sliced bam files containing the segments of interest. The first one should have ~200bp insertion, the second one has ~1kb deletion and the last one a ~40 bp deletion. I dont see any hint of these in the bam files, but just note an increased rate of misalignments in the second file where the mutation is supposed to be, and lack of reads spanning the deleted region in the 3rd file. So I was thinking maybe playing around with the alignment parameters should do the job.

Thank you for your time.

[out.zip](https://github.com/cauyrd/ScanIndel/files/2279465/out.zip)

**Comment by cauyrd** [comment]
I have trouble to create index file for the bam files you send. Below is
the error message. Can you send the index file and corrected bam file?

[bam_header_read] EOF marker is absent. The input is probably truncated.
[bam_header_read] invalid BAM binary header (this is not a BAM file).

**Comment by erfan1985** [comment]
Sorry about that. The files were in sam format. Please see the corrected and indexed files attached.
Just to add that I am not very confident that a sample switch hasn't occurred for the last file. But the first two are definitely correct.
Thanks

[corrected.zip](https://github.com/cauyrd/ScanIndel/files/2279713/corrected.zip)

**Comment by cauyrd** [comment]
Do you have the coordinate of the missed deletions in each of the sample?

**Comment by erfan1985** [comment]
bam_25 should have a 186bp insertion between chr12:133210818 and chr12:133210820. This is almost a duplication with just one base pair difference. Is ScanIndel good for detecting such insertions?

bam_26 should have a deletion from chr19:40902699 to chr19:40902776 (<100bp del)

These coordinates are found using Sanger. ScanIndel doesn't have to find the exact breakpoints I guess. 

Regarding the last file, I confirm now that it was a sample switch. So it should have nothing.

Thanks

**Comment by erfan1985** [comment]
Update:
I tried many many different tools for indel calling and none could identify these two variants except for Scalpel. It perfectly detected the deletion in bam_26, but still failed detecting the duplication in bam_25.

**Comment by cauyrd** [comment]
Scanindel can not detect duplication, but I tried one new unpublised tool
we are using, it detected the duplication as showed below for bam 25:

chr12 133210817 . . <TDUP> . .
NS=1;AO=23;DP=234;AB=0.1;SVLEN=198;SVTYPE=TDUP;SVMETHOD=ScanSV_ALN;CHR2=chr12;END=133211015
GT 1/1

Scanindel can detect a deletion in bam26 but this is the deletion you
mentioned, I believe.

chr19 55671134 . . <DEL> . . NS=1;AO=4;DP=17;AB=0.24;SVLEN=
37;SVTYPE=DEL;SVMETHOD=ScanSV_ALN;CHR2=chr19;END=55671171 GT 1/1

**Comment by erfan1985** [comment]
The duplication is correct, but the deletion is a different one. No other tools were able to detect it either. I tried GATK, pindel, VarDict, Delly, Manta, Lumpy, transIndel, fermikit, IMSindel and maybe more. Only Scalpel did!
What tool do you recommend for detection of duplication?

Thanks

**Comment by cauyrd** [comment]
Our in-house tool can do duplication. Have you tried pindel for this
duplication prediction?

**Comment by erfan1985** [comment]
Yes, Interestingly Pindel finds this duplication at low AF. Thanks for help
